examples/docker-sockshop.yml

```diff
@@ -0,0 +1,171 @@
+    networks:
+      - external
+  catalogue-db:
+    image: mysql
+    hostname: catalogue-db
+    dns: 172.17.0.1
+    dns_search: weave.local
+    restart: always
+    environment:
+      - reschedule:on-node-failure
+      - MYSQL_ROOT_PASSWORD=${MYSQL_ROOT_PASSWORD}
+      - MYSQL_ALLOW_EMPTY_PASSWORD=true
+      - MYSQL_DATABASE=socksdb
+    networks:
+      - external
+  cart:
+    image: weaveworksdemos/cart
+    hostname: cart
+    dns: 172.17.0.1
+    dns_search: weave.local
+    restart: always
+    environment:
+      - reschedule:on-node-failure
+    networks:
+      - internal
+  cart-db:
+    image: mongo
+    hostname: cart-db
+    dns: 172.17.0.1
+    dns_search: weave.local
+    restart: always
+    environment:
+      - reschedule:on-node-failure
+    networks:
+      - internal
+  orders:
+    image: weaveworksdemos/orders
+    hostname: orders
+    dns: 172.17.0.1
+    dns_search: weave.local
+    restart: always
+    environment:
+      - reschedule:on-node-failure
+    networks:
+      - internal
+      - secure
+      - backoffice
+  orders-db:
+    image: mongo
+    hostname: orders-db
+    dns: 172.17.0.1
+    dns_search: weave.local
+    restart: always
+    environment:
+      - reschedule:on-node-failure
+    networks:
+      - internal
+  shipping:
+    image: weaveworksdemos/shipping
+    hostname: shipping
+    dns: 172.17.0.1
+    dns_search: weave.local
+    restart: always
+    environment:
+      - reschedule:on-node-failure
+    networks:
+      - backoffice
+  queue-master:
+    image: weaveworksdemos/queue-master
+    hostname: queue-master
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
+    dns: 172.17.0.1
+    dns_search: weave.local
+    restart: always
+    environment:
+      - reschedule:on-node-failure
+    networks:
+      - backoffice
+  rabbitmq:
+    image: rabbitmq:3
+    hostname: rabbitmq
+    dns: 172.17.0.1
+    dns_search: weave.local
+    restart: always
+    environment:
+      - reschedule:on-node-failure
+    networks:
+      - backoffice
+  payment:
+    image: weaveworksdemos/payment
+    hostname: payment
+    dns: 172.17.0.1
+    dns_search: weave.local
+    restart: always
+    environment:
+      - reschedule:on-node-failure
+    networks:
+      - secure
+  user:
+    image: weaveworksdemos/user
+    hostname: user
+    dns: 172.17.0.1
+    dns_search: weave.local
+    restart: always
+    environment:
+      - MONGO_HOST=user-db:27017
+      - reschedule:on-node-failure
+    networks:
+      - secure
+  user-db:
+    image: weaveworksdemos/user-db
+    hostname: user-db
+    dns: 172.17.0.1
+    dns_search: weave.local
+    restart: always
+    environment:
+      - reschedule:on-node-failure
+    networks:
+      - secure
+  user-sim:
+    image: weaveworksdemos/load-test
+    hostname: user-simulator
+    domainname: weave.local
+    networks:
+      - external
+    command: "-d 60 -r 200 -c 2 -h edge-router"
+networks:
+  external:
+    driver: weavemesh
```

Comment:
Had to remove this driver spec on my laptop (don't have weave plugin) to get it to work with docker-compose.